pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +98,248 @@ func (builder *Builder) validate() (bool, error) {
+	}
+
+	builder.Definition.CreationTimestamp = metav1.Time{}
+
+	var err error
+	builder.Object, err = builder.apiClient.ConfigV1Interface.ClusterVersions().Update(
+		context.TODO(), builder.Definition, metav1.UpdateOptions{})
+
+	return builder, err
+}
+
+// Get fetches existing clusterversion from cluster.
+func (builder *Builder) Get() (*v1.ClusterVersion, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting existing clusterversion with name %s from cluster", builder.Definition.Name)
+
+	clusterVersionObj := &v1.ClusterVersion{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, clusterVersionObj)
+
+	if err != nil {
+		glog.V(100).Infof("Failed to get clusterversion object %s from cluster due to: %w",
+			builder.Definition.Name, err)
+
+		return nil, err
+	}
+
+	return clusterVersionObj, nil
+}
+
+// WaitUntilProgressing waits for timeout duration or until clusterversion is in Progressing state.
+func (builder *Builder) WaitUntilProgressing(timeout time.Duration) error {
+	return builder.WaitUntilConditionTrue("Progressing", timeout)
+}
+
+// WaitUntilAvailable waits for timeout duration or until clusterversion is in Available state.
+func (builder *Builder) WaitUntilAvailable(timeout time.Duration) error {
+	return builder.WaitUntilConditionTrue("Progressing", timeout)
+}
+
+// WaitUntilConditionTrue waits for timeout duration or until clusterversion gets to a specific status.
+func (builder *Builder) WaitUntilConditionTrue(
+	conditionType v1.ClusterStatusConditionType, timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	if !builder.Exists() {
+		return fmt.Errorf("%s clusterversion not found", builder.Definition.Name)
+	}
+
+	return wait.PollUntilContextTimeout(
+		context.TODO(), time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+			var err error
+			builder.Object, err = builder.Get()
+
+			if err != nil {
+				return false, nil
+			}
+
+			for _, condition := range builder.Object.Status.Conditions {
+				if condition.Type == conditionType {
+					return condition.Status == isTrue, nil
+				}
+			}
+
+			return false, err
+		})
+}
+
+// WaitUntilUpdateIsStarted waits until there is a history entry indicating the update start.
+func (builder *Builder) WaitUntilUpdateIsStarted(timeout time.Duration) error {
+	return builder.WaitUntilUpdateHistoryStateTrue(updateHistoryPartialState, timeout)
+}
+
+// WaitUntilUpdateIsCompleted waits until there is a history entry indicating the update completed.
+func (builder *Builder) WaitUntilUpdateIsCompleted(timeout time.Duration) error {
+	return builder.WaitUntilUpdateHistoryStateTrue(updateHistoryCompletedState, timeout)
+}
+
+// WaitUntilUpdateHistoryStateTrue waits until there is a history entry indicating an updateHistoryState.
+func (builder *Builder) WaitUntilUpdateHistoryStateTrue(
```

Comment:
Please check my comments for WaitUntilConditionTrue functions. Same comments are relevant here as well.

Reply:
Fixed, thanks